packages/shared/src/components/MainLayout.tsx

```diff
@@ -200,7 +177,22 @@ function MainLayoutComponent({
           isBannerAvailable && 'laptop:pt-8',
         )}
       >
-        <RenderSidebar />
+        {showSidebar && (
+          <Sidebar
+            promotionalBannerActive={isBannerAvailable}
+            sidebarRendered={sidebarRendered}
+            openMobileSidebar={openMobileSidebar}
+            onNavTabClick={onNavTabClick}
+            enableSearch={enableSearch}
+            activePage={activePage}
+            showDnd={showDnd}
+            dndActive={dndActive}
+            isNavButtons={isNavItemsButton}
+            onShowDndClick={onShowDndClick}
+            onLogoClick={onLogoClick}
+            setOpenMobileSidebar={() => onMobileSidebarToggle(false)}
+          />
+        )}
```

Comment:
I removed the `sidebarRendered === null` check, as `useSidebarRendered` either returns `true` or `false`. And I moved it down here, as something with it being in a const made it re-render often